### PR TITLE
Typecast pm2 env vars

### LINKED ISSUE #20375: Cannot set some PM2 configuration values
### Describe the Bug

Can't set some of the PM2 configuration variables since they are read in as string-type-literals.

For example, `PM2_AUTO_RESTART=true` will immediately crash with the following message:
```
2023-11-09T15:26:38: PM2 log: Launching in no daemon mode
2023-11-09T15:26:38: PM2 error: Expect "autorestart" to be a typeof [object Boolean], but now is [object String]
2023-11-09T15:26:39: PM2 log: PM2 successfully stopped
```

Seems like this is just because we load this variables outside of our env-casting magic.

### To Reproduce

```bash
docker run --rm -it \
    -e KEY=$(uuidgen) \
    -e SECRET=$(uuidgen) \
    -e PM2_AUTO_RESTART=true \
    directus/directus:10.7.2
```

### Directus Version

v10.7.x

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by DanielBiegler**
## Scope

What's changed:

- Typecasting for env vars

## Potential Risks / Drawbacks

- Potentially breaking the startup - worked for me though :pray: 

## Review Notes / Questions

- Thought to myself that we probably dont want to overcomplicate the config file with interpolation and object passing since if and when you actually need a specific setup, you'll probably be better off just creating your own ecosystem file. So I just went for simple typecasting to fix the urgent issue.

---

Fixes #20375

### CODE REVIEW

File: ecosystem.config.cjs [hunk 1]
```diff
@@ -12,17 +12,17 @@ module.exports = [
...
+		min_uptime: parseInt(process.env.PM2_MIN_UPTIME) || undefined,
+		listen_timeout: parseInt(process.env.PM2_LISTEN_TIMEOUT) || undefined,
+		kill_timeout: parseInt(process.env.PM2_KILL_TIMEOUT) || undefined,
 		wait_ready: true,
-		max_restarts: process.env.PM2_MAX_RESTARTS,
-		restart_delay: process.env.PM2_RESTART_DELAY ?? 0,
-		autorestart: process.env.PM2_AUTO_RESTART ?? false,
+		max_restarts: parseInt(process.env.PM2_MAX_RESTARTS) || undefined,
+		restart_delay: parseInt(process.env.PM2_RESTART_DELAY) || 0,
+		autorestart: ['1', 'true'].includes(String(process.env.PM2_AUTO_RESTART).toLowerCase()),
```
Comment (paescuj) [inline ecosystem.config.cjs:26]: I'd simplify this to
```suggestion
		autorestart: process.env.PM2_AUTO_RESTART === 'true',
```

File: ecosystem.config.cjs [hunk 2]
```diff
@@ -12,17 +12,17 @@ module.exports = [
...
-		kill_timeout: process.env.PM2_KILL_TIMEOUT,
+		min_uptime: parseInt(process.env.PM2_MIN_UPTIME) || undefined,
+		listen_timeout: parseInt(process.env.PM2_LISTEN_TIMEOUT) || undefined,
+		kill_timeout: parseInt(process.env.PM2_KILL_TIMEOUT) || undefined,
 		wait_ready: true,
-		max_restarts: process.env.PM2_MAX_RESTARTS,
-		restart_delay: process.env.PM2_RESTART_DELAY ?? 0,
-		autorestart: process.env.PM2_AUTO_RESTART ?? false,
+		max_restarts: parseInt(process.env.PM2_MAX_RESTARTS) || undefined,
+		restart_delay: parseInt(process.env.PM2_RESTART_DELAY) || 0,
```
Comment (paescuj) [inline ecosystem.config.cjs:25]: Do we intentionally fallback to `undefined` when values are `0`?

Comment (rijkvanzanten) [reply ecosystem.config.cjs:25]: I'm guessing that answer is _no_

Comment (DanielBiegler) [reply ecosystem.config.cjs:25]: Ohhhh _whoopsy_ youre right, thanks ... :sweat_smile:

File: ecosystem.config.cjs (hunk 1 above)
Comment (DanielBiegler) [reply ecosystem.config.cjs:26]: Alright, lets keep it simple :+1:

File: ecosystem.config.cjs (hunk 2 above)
Comment (DanielBiegler) [reply ecosystem.config.cjs]: I just reverted the number conversion to let pm2 handle those itself. At least this specific bug with the boolean is fixed and pm2 starts :+1:

**Review by paescuj — APPROVED** [review]

**Pull request merged**